src/oic/utils/authn/user.py

```diff
@@ -359,9 +356,6 @@ def __init__(self, srv, pwd, ttl=5):
     def verify_password(self, user, password):
         if user in self.passwd:
             _pwd = self.passwd[user]
-            if six.PY2:
-                if isinstance(_pwd, str):
-                    _pwd = _pwd.decode('utf-8')
             if _pwd != password:
```

Comment:
As this is Py3 now, we have a constant time compare (hmac.compare_digest()).

So maybe kill https://github.com/OpenIDC/pyoidc/issues/341 too by using it to compare the passwords.

Reply:
Good catch :) Will do that.